mk/spksrc.python.mk

```diff
@@ -0,0 +1,49 @@
+###
+### Reuse Python libraries
+###
+
+# Default python version to use
+ifeq ($(strip $(PYTHON_VERSION)),)
+export PYTHON_VERSION = 311
+endif
+
+# set default spk/python* path to use
+export PYTHON_DIR = $(realpath $(shell pwd)/../python$(PYTHON_VERSION)/work-$(ARCH)-$(TCVERSION)/install/var/packages/python$(PYTHON_VERSION)/target)
+
+# always define SPK_DEPENDS
+SPK_DEPENDS := "python$(PYTHON_VERSION)>=$(shell sed -n 's/^SPK_VERS = \(.*\)/\1/p' $(shell pwd)/../python$(PYTHON_VERSION)/Makefile)-$(shell sed -n 's/^SPK_REV = \(.*\)/\1/p' $(shell pwd)/../python$(PYTHON_VERSION)/Makefile)"
+
```

Comment:
IMHO SPK_DEPENDS must be defined by the respective package.

This would not work for packages that depend not only on python (like `SPK_DEPENDS = "python311:git"`)
And probably it is not wanted to include the python package revision in the dependency.

Reply:
I have a different view on this, at least food for thoughts...  I'm doing something in `tvheadend` that sort of solves this and would automate the populating of the SPK_DEPENDS variable value.

Something like the following in `spksrc.python.mk` would ensure python is always included in the SPK_DEPENDS:
```
PYTHON_DEPENDS = python$(PYTHON_VERSION)>=$(shell sed -n 's/^SPK_VERS = \(.*\)/\1/p' $(shell pwd)/../python$(PYTHON_VERSION)/Makefile)-$(shell sed -n 's/^SPK_REV = \(.*\)/\1/p' $(shell pwd)/../python$(PYTHON_VERSION)/Makefile)
ifneq ($(wildcard $(SPK_DEPENDS)),)
SPK_DEPENDS := "$(SPK_DEPENDS):$(PYTHON_DEPENDS)"
else
SPK_DEPENDS = "$(PYTHON_DEPENDS)"
endif
```